parlai/core/dialog_teacher.py

```diff
@@ -319,8 +319,17 @@ def share(self):
         return shared
 
     def __len__(self):
-        # unknown
-        return 0
+        length_file = self.datafile + ".length"
+        if not os.path.isfile(length_file):
+            length = 0
+            for episode in self._read_episode(self.data_loader(self.datafile)):
+                length += len(episode)
+            with open(length_file, 'w') as f:
+                f.write(str(length))
+        else:
+            with open(length_file, 'r') as f:
+                length = int(f.read())
+        return length
```

Comment:
if we've already loaded the data, then we can also get it just from there first rather than iterating over the data again

Reply:
nvm